### PR TITLE
CUMULUS-1040: Refactor ingest package

### PR DESCRIPTION
**Pull request by markdboyd**
- Refactor `ingest` package to use new `aws-client` and `errors` packages

Should merge after #1418

### CODE REVIEW

File: packages/ingest/aws.js [hunk 1]
```diff
@@ -48,340 +13,82 @@ function getEndpoint(local = false, port = 8000) {
...
-    };
-
-    const attr = await queue.getQueueAttributes(params).promise();
-    attr.Attributes.name = name;
-    return attr.Attributes;
-  }
-}
-
-
-class ECS {
```
Comment (markdboyd) [inline packages/ingest/aws.js]: This whole class wasn't used anywhere that I could see, so I removed it. Thoughts?

Comment (yjpa7145) [reply packages/ingest/aws.js]: +1 for removing it

File: packages/ingest/lock.js [hunk 2]
```diff
@@ -18,12 +22,12 @@ async function checkOldLocks(bucket, list) {
   if (list) {
     let count = list.length;
 
-    list.forEach((item) => {
+    list.forEach(async (item) => {
```
Comment (yjpa7145) [inline packages/ingest/lock.js]: There is a bug here, there is nothing waiting for each Promise to resolve.

File: packages/aws-client/index.js [hunk 3]
```diff
@@ -1,14 +1,12 @@
 exports.client = require('./client');
-exports.cloudformation = require('./cloudformation');
-exports.dynamo = require('./dynamo');
+exports.cloudformation = require('./CloudFormation');
```
Comment (yjpa7145) [inline packages/aws-client/index.js]: Seems like the exports in this file should match the case of the things that they're exporting. `s3` -> `S3`, for example.

File: packages/ingest/lock.js [hunk 4]
```diff
@@ -1,72 +1,73 @@
...
+* @param {Array} list - The list of locks in the bucket
+* @returns {integer} - Number of locks remaining in bucket
 **/
 async function checkOldLocks(bucket, list) {
-  if (list) {
-    let count = list.length;
-
-    list.forEach((item) => {
-      const date = list[item].LastModified;
+  if (list.length > 0) {
```
Comment (yjpa7145) [inline packages/ingest/lock.js]: Suggested refactor of this function:

```js
async function checkOldLocks(bucket, locks = []) {
  const fiveMinutesAgo = Date.now() - (5 * 60 * 1000);
  const expiredLocks = locks.filter((lock) => lock.LastModified < fiveMinutesAgo);
  await Promise.all(expiredLocks.map((lock) => deleteS3Object(bucket, lock.Key)));

  return locks.length - expiredLocks.length;
}
```

File: packages/ingest/lock.js [hunk 5]
```diff
@@ -1,72 +1,73 @@
...
+* Counts the number of locks in a bucket.
 *
 * @param {Object} bucket - The AWS S3 bucket to check
-* @param {string} pName - The provider name
+* @param {string} providerName - The provider name
 * @returns {integer} - Number of current locks in the bucket
 **/
-async function countLock(bucket, pName) {
-  const list = aws.s3().listObjectsV2({
+async function countLock(bucket, providerName) {
```
Comment (yjpa7145) [inline packages/ingest/lock.js]: 👍

**Pull request merged**